src/main/java/com/health/controller/HomeController.java

```diff
@@ -1016,6 +1026,83 @@ public String viewCoursesAvailable(HttpServletRequest req,
+
+        List<PackageEntity> packageEntitiesList = packageEntityService.findAll();
+        Collections.sort(packageEntitiesList);
+        model.addAttribute("packageEntitiesList", packageEntitiesList);
+
+        List<NptelTutorial> nptelTutorials = nptelTutorialService.findAll();
+        Collections.sort(nptelTutorials, NptelTutorial.SortByUploadTime);
+        model.addAttribute("nptelTutorials", nptelTutorials);
+        return "addNptelTutorial";
+    }
+
+    @PostMapping("/addNptelTutorial")
+    public String addNptelTutorialPost(@RequestParam(value = "add_csv_file") MultipartFile csv_file,
+            @RequestParam(name = "packageId") int packageId, @RequestParam(name = "packageName") String packageName,
+
+            HttpServletRequest req, Principal principal, Model model) {
+
+        User usr = getUser(principal);
+        logger.info("{} {} {}", usr.getUsername(), req.getMethod(), req.getRequestURI());
+        model.addAttribute("userInfo", usr);
+        PackageEntity packageEntity;
+
+        if (packageId == -1) {
+            packageEntity = packageEntityService.findByPackageName(packageName);
+        } else {
+            packageEntity = packageEntityService.findByPackageId(packageId);
+        }
+
+        if (packageEntity == null) {
+            packageEntity = new PackageEntity();
+            packageEntity.setPackageName(packageName);
+            packageEntity.setDateAdded(ServiceUtility.getCurrentTime());
+            packageEntityService.save(packageEntity);
+        }
+
+        Path csvFilePath = Paths.get(env.getProperty("spring.applicationexternalPath.name"),
+                CommonData.uploadNptelTutorial, "Sample_CSV_file_for_Nptel_Tutorial" + ".csv");
+        String temp = csvFilePath.toString();
+        int indexToStart = temp.indexOf("Media");
+        String document = temp.substring(indexToStart, temp.length());
+        model.addAttribute("sample_csv_file", document);
+
+        try {
+            nptelTutorialService.saveNptelTutorialsFromCSV(csv_file, model, packageEntity);
+            List<PackageEntity> packageEntitiesList = packageEntityService.findAll();
+            Collections.sort(packageEntitiesList);
+            model.addAttribute("packageEntitiesList", packageEntitiesList);
+
+        } catch (IOException e) {
+
+            model.addAttribute("error_msg", "Some Errors Occured Please contact Admin or try again");
+            logger.error("Exception: ", e);
+            return "addNptelTutorial";
+        } catch (CsvException e) {
+            model.addAttribute("error_msg", "Some Errors Occured Please contact Admin or try again");
+            logger.error("Exception: ", e);
+            return "addNptelTutorial";
+        }
+
+        List<NptelTutorial> nptelTutorials = nptelTutorialService.findAll();
+        Collections.sort(nptelTutorials, NptelTutorial.SortByUploadTime);
+        model.addAttribute("nptelTutorials", nptelTutorials);
+
+        return "addNptelTutorial";

```

Comment:
```suggestion
        return  addNptelTutorial(req, principal, model);
```
Replace with addNptelTutorial() call. Remove code which is already there in addNptelTutorial()